### ISSUE TITLE
Security: Rebuild binaries and Docker images after Go fixed CVEs

### ISSUE THREAD

**Issue by reneleonhardt** [cybertec-postgresql/pgwatch2#796]
### Security
Your last binaries and Docker images contain no fixes for latest vulnerabilities (built with Go 1.22.3):
* Critical 9.8: CVE-2024-24790
* Fixed in Go 1.22.4: https://github.com/golang/go/issues/67682
* https://go.dev/doc/devel/release#go1.22.4

Please rebuild everything to make trivy happy again 😄

**Comment by pashagolub** [comment]
Hello.

https://hub.docker.com/r/cybertec/pgwatch2-postgres/ latest is built 12 days ago, which means it's up to date. :)

What is trivy and why should we care?

**Comment by pashagolub** [comment]
Closed as staled

**Comment by reneleonhardt** [comment]
You don't have to care if your users shouldn't use your version tags because you only want to rebuild `latest` after security fixes are available, for example the fixes for `net/http` in https://go.dev/doc/devel/release#go1.22.5.
But then please change your documentation accordingly, otherwise users expect a version rebuild (new image digest, same tag `1.13.0` like you do with `latest`) or a new patch version (new tag `1.13.1`), then automatic updaters like Dependabot or Renovate could automatically suggest or merge your changed image.

https://blog.devops.dev/mastering-trivy-your-ultimate-guide-to-securing-containers-and-artifacts-in-devops-77324613aaad

```
$ docker pull cybertec/pgwatch2-daemon:1.13.0
$ trivy image cybertec/pgwatch2-daemon:1.13.0

cybertec/pgwatch2-daemon:1.13.0 (ubuntu 22.04)

Total: 70 (UNKNOWN: 0, LOW: 41, MEDIUM: 28, HIGH: 1, CRITICAL: 0)

┌──────────────────┬──────────────────┬──────────┬──────────┬──────────────────────────┬──────────────────────┬──────────────────────────────────────────────────────────────┐
│     Library      │  Vulnerability   │ Severity │  Status  │    Installed Version     │    Fixed Version     │                            Title                             │
├──────────────────┼──────────────────┼──────────┼──────────┼──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ coreutils        │ CVE-2016-2781    │ LOW      │ affected │ 8.32-4.1ubuntu1.2        │                      │ coreutils: Non-privileged session can escape to the parent   │
│                  │                  │          │          │                          │                      │ session in chroot                                            │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2016-2781                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ gcc-12-base      │ CVE-2022-27943   │          │          │ 12.3.0-1ubuntu1~22.04    │                      │ binutils: libiberty/rust-demangle.c in GNU GCC 11.2 allows   │
│                  │                  │          │          │                          │                      │ stack exhaustion in demangle_const                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-27943                   │
├──────────────────┼──────────────────┼──────────┼──────────┼──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ git              │ CVE-2024-32002   │ MEDIUM   │ fixed    │ 1:2.34.1-1ubuntu1.10     │ 1:2.34.1-1ubuntu1.11 │ git: Recursive clones RCE                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32002                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32004   │          │          │                          │                      │ git: RCE while cloning local repos                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32004                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32020   │          │          │                          │                      │ git: insecure hardlinks                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32020                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32021   │          │          │                          │                      │ git: symlink bypass                                          │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32021                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32465   │          │          │                          │                      │ git: additional local RCE                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32465                   │
│                  ├──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2018-1000021 │ LOW      │ affected │                          │                      │ git: client prints server-sent ANSI escape codes to the      │
│                  │                  │          │          │                          │                      │ terminal, allowing for...                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2018-1000021                 │
├──────────────────┼──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ git-man          │ CVE-2024-32002   │ MEDIUM   │ fixed    │                          │ 1:2.34.1-1ubuntu1.11 │ git: Recursive clones RCE                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32002                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32004   │          │          │                          │                      │ git: RCE while cloning local repos                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32004                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32020   │          │          │                          │                      │ git: insecure hardlinks                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32020                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32021   │          │          │                          │                      │ git: symlink bypass                                          │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32021                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-32465   │          │          │                          │                      │ git: additional local RCE                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-32465                   │
│                  ├──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2018-1000021 │ LOW      │ affected │                          │                      │ git: client prints server-sent ANSI escape codes to the      │
│                  │                  │          │          │                          │                      │ terminal, allowing for...                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2018-1000021                 │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ gpgv             │ CVE-2022-3219    │          │          │ 2.2.27-3ubuntu2.1        │                      │ gnupg: denial of service issue (resource consumption) using  │
│                  │                  │          │          │                          │                      │ compressed packets                                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-3219                    │
├──────────────────┼──────────────────┼──────────┼──────────┼──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libc-bin         │ CVE-2024-33599   │ MEDIUM   │ fixed    │ 2.35-0ubuntu3.7          │ 2.35-0ubuntu3.8      │ glibc: stack-based buffer overflow in netgroup cache         │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33599                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-33600   │          │          │                          │                      │ glibc: null pointer dereferences after failed netgroup cache │
│                  │                  │          │          │                          │                      │ insertion                                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33600                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-33601   │          │          │                          │                      │ glibc: netgroup cache may terminate daemon on memory         │
│                  │                  │          │          │                          │                      │ allocation failure                                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33601                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-33602   │          │          │                          │                      │ glibc: netgroup cache assumes NSS callback uses in-buffer    │
│                  │                  │          │          │                          │                      │ strings                                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33602                   │
│                  ├──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2016-20013   │ LOW      │ affected │                          │                      │ sha256crypt and sha512crypt through 0.6 allow attackers to   │
│                  │                  │          │          │                          │                      │ cause a denial of...                                         │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2016-20013                   │
├──────────────────┼──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libc6            │ CVE-2024-33599   │ MEDIUM   │ fixed    │                          │ 2.35-0ubuntu3.8      │ glibc: stack-based buffer overflow in netgroup cache         │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33599                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-33600   │          │          │                          │                      │ glibc: null pointer dereferences after failed netgroup cache │
│                  │                  │          │          │                          │                      │ insertion                                                    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33600                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-33601   │          │          │                          │                      │ glibc: netgroup cache may terminate daemon on memory         │
│                  │                  │          │          │                          │                      │ allocation failure                                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33601                   │
│                  ├──────────────────┤          │          │                          │                      ├──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-33602   │          │          │                          │                      │ glibc: netgroup cache assumes NSS callback uses in-buffer    │
│                  │                  │          │          │                          │                      │ strings                                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-33602                   │
│                  ├──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2016-20013   │ LOW      │ affected │                          │                      │ sha256crypt and sha512crypt through 0.6 allow attackers to   │
│                  │                  │          │          │                          │                      │ cause a denial of...                                         │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2016-20013                   │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libgcc-s1        │ CVE-2022-27943   │          │          │ 12.3.0-1ubuntu1~22.04    │                      │ binutils: libiberty/rust-demangle.c in GNU GCC 11.2 allows   │
│                  │                  │          │          │                          │                      │ stack exhaustion in demangle_const                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-27943                   │
├──────────────────┼──────────────────┼──────────┤          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libgcrypt20      │ CVE-2024-2236    │ MEDIUM   │          │ 1.9.4-3ubuntu3           │                      │ libgcrypt: vulnerable to Marvin Attack                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-2236                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libgssapi-krb5-2 │ CVE-2024-26462   │          │          │ 1.19.2-2ubuntu0.3        │                      │ krb5: Memory leak at /krb5/src/kdc/ndr.c                     │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26462                   │
│                  ├──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26458   │ LOW      │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/rpc/pmap_rmt.c            │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26458                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26461   │          │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/gssapi/krb5/k5sealv3.c    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26461                   │
├──────────────────┼──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libk5crypto3     │ CVE-2024-26462   │ MEDIUM   │          │                          │                      │ krb5: Memory leak at /krb5/src/kdc/ndr.c                     │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26462                   │
│                  ├──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26458   │ LOW      │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/rpc/pmap_rmt.c            │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26458                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26461   │          │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/gssapi/krb5/k5sealv3.c    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26461                   │
├──────────────────┼──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libkrb5-3        │ CVE-2024-26462   │ MEDIUM   │          │                          │                      │ krb5: Memory leak at /krb5/src/kdc/ndr.c                     │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26462                   │
│                  ├──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26458   │ LOW      │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/rpc/pmap_rmt.c            │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26458                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26461   │          │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/gssapi/krb5/k5sealv3.c    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26461                   │
├──────────────────┼──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libkrb5support0  │ CVE-2024-26462   │ MEDIUM   │          │                          │                      │ krb5: Memory leak at /krb5/src/kdc/ndr.c                     │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26462                   │
│                  ├──────────────────┼──────────┤          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26458   │ LOW      │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/rpc/pmap_rmt.c            │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26458                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-26461   │          │          │                          │                      │ krb5: Memory leak at /krb5/src/lib/gssapi/krb5/k5sealv3.c    │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-26461                   │
├──────────────────┼──────────────────┼──────────┤          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ liblzma5         │ CVE-2020-22916   │ MEDIUM   │          │ 5.2.5-2ubuntu1           │                      │ Denial of service via decompression of crafted file          │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2020-22916                   │
├──────────────────┼──────────────────┼──────────┤          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libncurses6      │ CVE-2023-45918   │ LOW      │          │ 6.3-2ubuntu0.1           │                      │ ncurses 6.4-20230610 has a NULL pointer dereference in       │
│                  │                  │          │          │                          │                      │ tgetstr in tinf ......                                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-45918                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2023-50495   │          │          │                          │                      │ ncurses: segmentation fault via _nc_wrap_entry()             │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-50495                   │
├──────────────────┼──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libncursesw6     │ CVE-2023-45918   │          │          │                          │                      │ ncurses 6.4-20230610 has a NULL pointer dereference in       │
│                  │                  │          │          │                          │                      │ tgetstr in tinf ......                                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-45918                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2023-50495   │          │          │                          │                      │ ncurses: segmentation fault via _nc_wrap_entry()             │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-50495                   │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libpcre3         │ CVE-2017-11164   │          │          │ 2:8.39-13ubuntu0.22.04.1 │                      │ pcre: OP_KETRMAX feature in the match function in            │
│                  │                  │          │          │                          │                      │ pcre_exec.c                                                  │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2017-11164                   │
├──────────────────┼──────────────────┼──────────┼──────────┼──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libssl3          │ CVE-2022-40735   │ MEDIUM   │ fixed    │ 3.0.2-0ubuntu1.15        │ 3.0.2-0ubuntu1.16    │ The Diffie-Hellman Key Agreement Protocol allows use of long │
│                  │                  │          │          │                          │                      │ exponents that arguably...                                   │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-40735                   │
│                  ├──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-2511    │ LOW      │ affected │                          │                      │ openssl: Unbounded memory growth with session handling in    │
│                  │                  │          │          │                          │                      │ TLSv1.3                                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-2511                    │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-4603    │          │          │                          │                      │ openssl: Excessive time spent checking DSA keys and          │
│                  │                  │          │          │                          │                      │ parameters                                                   │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-4603                    │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-4741    │          │          │                          │                      │ openssl: Use After Free with SSL_free_buffers                │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-4741                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libstdc++6       │ CVE-2022-27943   │          │          │ 12.3.0-1ubuntu1~22.04    │                      │ binutils: libiberty/rust-demangle.c in GNU GCC 11.2 allows   │
│                  │                  │          │          │                          │                      │ stack exhaustion in demangle_const                           │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-27943                   │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libsystemd0      │ CVE-2023-7008    │          │          │ 249.11-0ubuntu3.12       │                      │ systemd-resolved: Unsigned name response in signed zone is   │
│                  │                  │          │          │                          │                      │ not refused when DNSSEC=yes...                               │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-7008                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libtinfo6        │ CVE-2023-45918   │          │          │ 6.3-2ubuntu0.1           │                      │ ncurses 6.4-20230610 has a NULL pointer dereference in       │
│                  │                  │          │          │                          │                      │ tgetstr in tinf ......                                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-45918                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2023-50495   │          │          │                          │                      │ ncurses: segmentation fault via _nc_wrap_entry()             │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-50495                   │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libudev1         │ CVE-2023-7008    │          │          │ 249.11-0ubuntu3.12       │                      │ systemd-resolved: Unsigned name response in signed zone is   │
│                  │                  │          │          │                          │                      │ not refused when DNSSEC=yes...                               │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-7008                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ libzstd1         │ CVE-2022-4899    │          │          │ 1.4.8+dfsg-3build1       │                      │ zstd: mysql: buffer overrun in util.c                        │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-4899                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ login            │ CVE-2023-29383   │          │          │ 1:4.8.1-2ubuntu2.2       │                      │ shadow: Improper input validation in shadow-utils package    │
│                  │                  │          │          │                          │                      │ utility chfn                                                 │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-29383                   │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ ncurses-base     │ CVE-2023-45918   │          │          │ 6.3-2ubuntu0.1           │                      │ ncurses 6.4-20230610 has a NULL pointer dereference in       │
│                  │                  │          │          │                          │                      │ tgetstr in tinf ......                                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-45918                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2023-50495   │          │          │                          │                      │ ncurses: segmentation fault via _nc_wrap_entry()             │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-50495                   │
├──────────────────┼──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│ ncurses-bin      │ CVE-2023-45918   │          │          │                          │                      │ ncurses 6.4-20230610 has a NULL pointer dereference in       │
│                  │                  │          │          │                          │                      │ tgetstr in tinf ......                                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-45918                   │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2023-50495   │          │          │                          │                      │ ncurses: segmentation fault via _nc_wrap_entry()             │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-50495                   │
├──────────────────┼──────────────────┼──────────┼──────────┼──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ openssh-client   │ CVE-2024-6387    │ HIGH     │ fixed    │ 1:8.9p1-3ubuntu0.7       │ 1:8.9p1-3ubuntu0.10  │ openssh: Possible remote code execution due to a race        │
│                  │                  │          │          │                          │                      │ condition in signal...                                       │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-6387                    │
├──────────────────┼──────────────────┼──────────┤          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ openssl          │ CVE-2022-40735   │ MEDIUM   │          │ 3.0.2-0ubuntu1.15        │ 3.0.2-0ubuntu1.16    │ The Diffie-Hellman Key Agreement Protocol allows use of long │
│                  │                  │          │          │                          │                      │ exponents that arguably...                                   │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2022-40735                   │
│                  ├──────────────────┼──────────┼──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-2511    │ LOW      │ affected │                          │                      │ openssl: Unbounded memory growth with session handling in    │
│                  │                  │          │          │                          │                      │ TLSv1.3                                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-2511                    │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-4603    │          │          │                          │                      │ openssl: Excessive time spent checking DSA keys and          │
│                  │                  │          │          │                          │                      │ parameters                                                   │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-4603                    │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-4741    │          │          │                          │                      │ openssl: Use After Free with SSL_free_buffers                │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-4741                    │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ passwd           │ CVE-2023-29383   │          │          │ 1:4.8.1-2ubuntu2.2       │                      │ shadow: Improper input validation in shadow-utils package    │
│                  │                  │          │          │                          │                      │ utility chfn                                                 │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2023-29383                   │
├──────────────────┼──────────────────┤          │          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ patch            │ CVE-2018-6952    │          │          │ 2.7.6-7build2            │                      │ patch: Double free of memory in pch.c:another_hunk() causes  │
│                  │                  │          │          │                          │                      │ a crash                                                      │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2018-6952                    │
│                  ├──────────────────┤          │          │                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2021-45261   │          │          │                          │                      │ patch: Invalid Pointer via another_hunk function             │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2021-45261                   │
├──────────────────┼──────────────────┼──────────┤          ├──────────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────┤
│ wget             │ CVE-2021-31879   │ MEDIUM   │          │ 1.21.2-2ubuntu1          │                      │ wget: authorization header disclosure on redirect            │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2021-31879                   │
│                  ├──────────────────┤          ├──────────┤                          ├──────────────────────┼──────────────────────────────────────────────────────────────┤
│                  │ CVE-2024-38428   │          │ fixed    │                          │ 1.21.2-2ubuntu1.1    │ wget: Misinterpretation of input may lead to improper        │
│                  │                  │          │          │                          │                      │ behavior                                                     │
│                  │                  │          │          │                          │                      │ https://avd.aquasec.com/nvd/cve-2024-38428                   │
└──────────────────┴──────────────────┴──────────┴──────────┴──────────────────────────┴──────────────────────┴──────────────────────────────────────────────────────────────┘

pgwatch2/pgwatch2 (gobinary)

Total: 4 (UNKNOWN: 1, LOW: 0, MEDIUM: 2, HIGH: 0, CRITICAL: 1)

┌─────────────────┬────────────────┬──────────┬────────┬──────────────────────┬─────────────────┬──────────────────────────────────────────────────────────────┐
│     Library     │ Vulnerability  │ Severity │ Status │  Installed Version   │  Fixed Version  │                            Title                             │
├─────────────────┼────────────────┼──────────┼────────┼──────────────────────┼─────────────────┼──────────────────────────────────────────────────────────────┤
│ go.etcd.io/etcd │ CVE-2018-1099  │ MEDIUM   │ fixed  │ v3.3.27+incompatible │ 3.4.0           │ etcd: DNS rebinding vulnerability in etcd server             │
│                 │                │          │        │                      │                 │ https://avd.aquasec.com/nvd/cve-2018-1099                    │
├─────────────────┼────────────────┼──────────┤        ├──────────────────────┼─────────────────┼──────────────────────────────────────────────────────────────┤
│ stdlib          │ CVE-2024-24790 │ CRITICAL │        │ 1.22.3               │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for   │
│                 │                │          │        │                      │                 │ IPv4-mapped IPv6 addresses                                   │
│                 │                │          │        │                      │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                   │
│                 ├────────────────┼──────────┤        │                      │                 ├──────────────────────────────────────────────────────────────┤
│                 │ CVE-2024-24789 │ MEDIUM   │        │                      │                 │ golang: archive/zip: Incorrect handling of certain ZIP files │
│                 │                │          │        │                      │                 │ https://avd.aquasec.com/nvd/cve-2024-24789                   │
│                 ├────────────────┼──────────┤        │                      ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                 │ CVE-2024-24791 │ UNKNOWN  │        │                      │ 1.21.12, 1.22.5 │ The net/http HTTP/1.1 client mishandled the case where a     │
│                 │                │          │        │                      │                 │ server responds to...                                        │
│                 │                │          │        │                      │                 │ https://avd.aquasec.com/nvd/cve-2024-24791                   │
└─────────────────┴────────────────┴──────────┴────────┴──────────────────────┴─────────────────┴──────────────────────────────────────────────────────────────┘
```
